hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java

```diff
@@ -167,8 +175,16 @@ protected boolean submitOp(HapiSpec spec) {
                                                 : MAX_OPS_PER_SEC - opsThisSecond.get()))
                         .mapToObj(ignore -> provider.get())
                         .flatMap(Optional::stream)
-                        .filter(op -> op != Stream.empty())
-                        .peek(op -> counts.get(op.type()).getAndIncrement())
+                        .peek(op -> {
+                            counts.get(op.type()).getAndIncrement();
+                            if (loggingOff) {
+                                if (op instanceof HapiTxnOp<?> txnOp) {
+                                    txnOp.noLogging();
```

Comment:
Why isn't `noLogging()` (and associated getter) part of parent `HapiSpecOperation`?

Reply:
Type inference for method chaining only works one level deep in the inheritance hierarchy; if we pull `noLogging()` up into `HapiSpecOperation`, then its return type would not be inferred as e.g. `HapiCryptoTransfer`.

Reply:
Eeeee.... ok.